include/ace/managers/state.h

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef _ACE_MANAGERS_STATE_H_
+#define _ACE_MANAGERS_STATE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <ace/types.h> // Amiga typedefs
+
+/* Types */
+
+typedef void (*tStateCb)(void);
+
+typedef struct _tState {
```

Comment:
You should document which struct fields are mandatory when creating state struct by hand, and all special rules. For instance, should pPrev be set to zero? Or is the value irrelevant and will be reset by push()?

EDIT: now I understand that you should use stateCreate(). This should be pointed in doxy for struct.